python/pyarrow/tests/test_parquet.py

```diff
@@ -685,6 +685,30 @@ def test_compare_schemas():
     assert fileh.schema[0] != 'arbitrary object'
 
 
+def test_validate_schema_write_table(tmpdir):
+    # ARROW-2926
+    import pyarrow.parquet as pq
+
+    simple_fields = [
+        pa.field('POS', pa.uint32()),
+        pa.field('desc', pa.string())
+    ]
+
+    simple_schema = pa.schema(simple_fields)
+
+    # simple_table schema does not match simple_schema
+    simple_from_array = [pa.array([1]), pa.array(['bla'])]
```

Comment:
Is this invalid because the array type is inferred to be `int64`?

Reply:
yep